floater/generators.py

```diff
@@ -70,55 +79,11 @@ def get_rectmesh(self):
             2D array of float y coordinates
         """
 
-        return np.meshgrid(self.x, self.y)
-
-    def get_oceancoords(self, model_grid, mesh='rect'):
-        """Get the coordinates of float positions taking into account the land mask.
-
-        PARAMETERS
-        ----------
-        model_grid : dictionary 
-            the following key value pairs are expected  
-                'land_mask': np.ndarray of bools
-                    2d array of dimensions len(lon) by len(lat).
-                    An element is True iff the corresponding tracer cell grid point is unmasked (ocean)
-                'lon': 1d array of the mask grid longitudes      
-                'lat': 1d array of the mask grid latitudes
-                     
-         mesh : string
-            options are 
-                'rect' - a rectangular mesh, the default 
-                'hex' - a hexagonal mesh 
-
-
-        RETURNS
-        -------
-        floats_ocean: np.ndarray
-            1D array of float coordinate subarrays:
-            e.g. floats_ocean[i] = [some_float_lon, some_float_lat] 
-        """
-
-        xx, yy = self.get_rectmesh()
+        xx, yy = np.meshgrid(self.x, self.y)
+        if self.model_grid is not None:
+            xx, yy = _subset_floats_from_mask(xx, yy, self.model_grid)
```

Comment:
when outputting a mesh, we check for the presence of a model grid and, if necessary, apply the land mask by calling ``_subset_floats_from_mask``, a standalone function.